### ISSUE TITLE
[FEATURE] Request for more actionable error messages

### ISSUE THREAD

**Issue by marlee114** [SaptarshiSarkar12/Drifty#213]
### Is your feature request related to a problem? Please describe.

When using the Drifty application, error messages displayed to the user do not provide much aid in helping them troubleshoot or understand what may be the problem with a certain download request. 

### Describe the solution you'd like.

I would like the application to provide more actionable error messages that are descriptive, and in plain english. As a result, the user may be able to fix a mistake in their request, or at least understand what may be causing the download to fail. 

### Screenshots

![test11](https://github.com/SaptarshiSarkar12/Drifty/assets/71605411/36e09234-c2e0-4429-abc1-80d2caa72179)
![test4-spy csv-file](https://github.com/SaptarshiSarkar12/Drifty/assets/71605411/88c1c715-5ad5-4830-85b1-974d16ceb5d8)

**Comment by SaptarshiSarkar12** [comment]
Also, please check the latest code by running the application in an IDE (preferably [IntelliJ Idea Community Edition](https://www.jetbrains.com/idea/download/)). 
**NOTE** : Please run ```Drifty_GUI``` file to start the _GUI_ and ```Drifty_CLI``` file to run the _CLI_. You can find those java programs in the **GUI** and **CLI** directory under ```src/main``` directory.

**Comment by rohanshah18** [comment]
Can you assign this to me please?

**Comment by marlee114** [comment]
> 

Rohan can have it, if he wants

**Comment by marlee114** [comment]
> Also, please check the latest code by running the application in an IDE (preferably [IntelliJ Idea Community Edition](https://www.jetbrains.com/idea/download/)).
> **NOTE** : Please run `Drifty_GUI` file to start the _GUI_ and `Drifty_CLI` file to run the _CLI_. You can find those java programs in the **GUI** and **CLI** directory under `src/main` directory.

I ran it on the windows command line, but maybe on an IDE its different

**Comment by SaptarshiSarkar12** [comment]
> Can you assign this to me please?

@rohanshah18 Yes 👍. You can work on it as @marlee114 has no problem.

**Comment by rohanshah18** [comment]
Thanks @marlee114! I'll start working on this and have an update by weekend.
